tutorial.md

```diff
@@ -0,0 +1,131 @@
+<!---
+Note: This tutorial is meant for Google Cloud Shell, and can be opened by going to
+https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/magic-modules&tutorial=tutorial.md
+--->
+# Magic Modules Tutorial
+
+## Intro
+
+This tutorial will walk you through the components that make up Magic Modules.
+
+## api.yaml
+
+Each product's api definition is stored in the magic-modules repo.
+
+Let's open
+<walkthrough-editor-open-file filePath="magic-modules/products/compute/api.yaml"
+                              text="products/compute/api.yaml">
+</walkthrough-editor-open-file>.
+
+### Product metadata
+
+The
+<walkthrough-editor-select-regex filePath="magic-modules/products/compute/api.yaml"
+                                 regex="!ruby/object:Api::Product"
+                                 text="top section">
+</walkthrough-editor-select-regex>
+provides metadata about the API, such as name, scopes, and versions.
+
+### Resources
+
+Each `api.yaml` file contains a list of resources.
```

Comment:
Might be worth adding what a 'resource' is in GCP.

Reply:
+1